lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4130,6 +4130,10 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
         : nullptr;
     Lowering::GenericContextScope genericContextScope(
         getContext().getTypeConverter(), adjGenSig);
+    auto origExitIt = original.findReturnBB();
+    assert(!origExitIt.isEnd() &&
```

Comment:
`isEnd()` doesn't seem to work by default:
```
/home/swiftci/jenkins/workspace/swift-PR-TensorFlow-Linux/llvm/include/llvm/ADT/ilist_node.h:101:5: error: static_assert failed due to requirement 'node_options<SILBasicBlock, true, false, void>::is_sentinel_tracking_explicit' "Use ilist_sentinel_tracking<true> to enable isSentinel()"
    static_assert(OptionsT::is_sentinel_tracking_explicit,
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/swiftci/jenkins/workspace/swift-PR-TensorFlow-Linux/llvm/include/llvm/ADT/ilist_iterator.h:177:50: note: in instantiation of member function 'llvm::ilist_node_impl<llvm::ilist_detail::node_options<swift::SILBasicBlock, true, false, void> >::isSentinel' requested here
  bool isEnd() const { return NodePtr ? NodePtr->isSentinel() : false; }
                                                 ^
/home/swiftci/jenkins/workspace/swift-PR-TensorFlow-Linux/swift/lib/SILOptimizer/Mandatory/Differentiation.cpp:4134:24: note: in instantiation of member function 'llvm::ilist_iterator<llvm::ilist_detail::node_options<swift::SILBasicBlock, true, false, void>, false, false>::isEnd' requested here
    assert(!origExitIt.isEnd() &&
                       ^
```
I dunno what's the best fix though.